docs/DeveloperGuide.md

```diff
@@ -6,7 +6,134 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+Our program is structured as pictured in the following class diagram:
+![ClassDiagram.png](diagrams/ClassDiagram.png)
```

Comment:
Consider changing the following:
1. Avoid overlapping arrows
2. Include the navigability between `Duke` and `moduleList`, `Duke` and `UI`
3. Include the visibility of the variables and methods